core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/DefaultEncodingFactory.java

```diff
@@ -216,8 +216,13 @@ private static DataType fitDelta(DataType dataType, Object max, Object min) {
     } else if (dataType == DataTypes.INT) {
       value = (long) (int) max - (long) (int) min;
```

Comment:
Even for all other datatypes has same problem right, I mean there is a chance of giving bigger datatype than original type.

Reply:
As per our discussion for other dataTypes overflow handling not required.